CONTRIBUTING.md

```diff
@@ -73,226 +73,126 @@ We'll lay out three examples below (one that uses our format for messages, one t
-            expEpochInfo:         types.EpochInfo{Identifier: "hello", Duration: time.Minute, StartTime: block1Time, CurrentEpoch: 2, CurrentEpochStartTime: block1Time.Add(time.Minute), CurrentEpochStartHeight: 3},
-        },
-        "StartTime in future won't get ticked on first block": {
-            initialEpochInfo: types.EpochInfo{StartTime: block1Time.Add(time.Second), CurrentEpoch: 0, CurrentEpochStartTime: time.Time{}},
-            // currentEpochStartHeight is 1 because thats when the timer was created on-chain
-            expEpochInfo: types.EpochInfo{StartTime: block1Time.Add(time.Second), CurrentEpoch: 0, CurrentEpochStartTime: time.Time{}, CurrentEpochStartHeight: 1},
+// TestMintExportGenesis tests that genesis is exported correctly.
+// It first initializes genesis to the expected value. Then, attempts
+// to export it. Lastly, compares exported to the expected.
+func(suite *KeeperTestSuite) TestMintExportGenesis() {
+    testCases: = map[string] struct {
+        expectedGenesis *types.GenesisState
+    } {
+        "default genesis": {
+            expectedGenesis: types.DefaultGenesisState(),
         },
-        "StartTime in past will get ticked on first block": {
-            initialEpochInfo: types.EpochInfo{StartTime: block1Time.Add(-time.Second), CurrentEpoch: 0, CurrentEpochStartTime: time.Time{}},
-            expEpochInfo:     types.EpochInfo{StartTime: block1Time.Add(-time.Second), CurrentEpoch: 1, CurrentEpochStartTime: block1Time.Add(-time.Second), CurrentEpochStartHeight: 1},
+        "custom genesis": {
+            expectedGenesis: customGenesis,
         },
     }
-    for name, test := range tests {
+
+    for name, tc: = range testCases {
         suite.Run(name, func() {
-            suite.SetupTest()
-            suite.Ctx = suite.Ctx.WithBlockHeight(1).WithBlockTime(block1Time)
-            initialEpoch := initializeBlankEpochInfoFields(test.initialEpochInfo, defaultIdentifier, defaultDuration)
-            suite.App.EpochsKeeper.AddEpochInfo(suite.Ctx, initialEpoch)
-            suite.App.EpochsKeeper.BeginBlocker(suite.Ctx)
-
-            // get sorted heights
-            heights := maps.Keys(test.blockHeightTimePairs)
-            osmoutils.SortSlice(heights)
-            for _, h := range heights {
-                // for each height in order, run begin block
-                suite.Ctx = suite.Ctx.WithBlockHeight(int64(h)).WithBlockTime(test.blockHeightTimePairs[h])
-                suite.App.EpochsKeeper.BeginBlocker(suite.Ctx)
-            }
-            expEpoch := initializeBlankEpochInfoFields(test.expEpochInfo, initialEpoch.Identifier, initialEpoch.Duration)
-            actEpoch := suite.App.EpochsKeeper.GetEpochInfo(suite.Ctx, initialEpoch.Identifier)
-            suite.Require().Equal(expEpoch, actEpoch)
+            // Setup.
+            app: = suite.App
+            ctx: = suite.Ctx
```

Comment:
In my opinion, we should be encouraging doing setup directly in the body of the for loop like it is here